src/js/firstDiscoveryEditor.js

```diff
@@ -1,267 +1,411 @@
+        },  {
+            source: "{that}.options.outerUiEnhancerGrades",
+            target: "{that enhancer}.options.gradeNames"
+        } ]
+    });
+
+
+    // Parses the "preview" variable out of the querystring, returning the preview page to use
+    gpii.firstDiscovery.getPreviewName = function () {
+        var previewPage = "searchpreview.html"; //default to search
+        var queryKvps = gpii.firstDiscovery.getQueryStringKeys(window.location.search);
+        if (queryKvps.preview && queryKvps.preview != "undefined") {
+            previewPage = queryKvps["preview"];
+        }
+        return previewPage;
+    };
+
+    gpii.firstDiscovery.getQueryStringKeys = function (querystring) {
+        var queryKvps = {};
+        var kvps = querystring.substring(1).split("&");
+        for (var i = 0; i < kvps.length; i++) {
+            var kvp_pieces = kvps[i].split("=");
+            queryKvps[decodeURIComponent(kvp_pieces[0])] = decodeURIComponent(kvp_pieces[1]);
+        }
+        return queryKvps;
+    };
+
+    gpii.firstDiscovery.getPreviewUrl = function (that, languageCode) {
+        var baseurl = "../../src/html/";
+        var previewname = gpii.firstDiscovery.getPreviewName();
+        var previewpage = "searchpreview.html"; //default
+        if (previewname === "search") {
+            previewpage = "searchpreview.html";
+        } else if (previewname === "electron") {
+            previewpage = "electronpreview.html";
+        }
+
+        //unsupported languages will return english preview.
+        //TODO: create previews for remaining languages, de-DE, nl-NL, sv-SE
+        if(languageCode == "de-DE"){
+            languageCode = "en-US";
+        }
+        if(languageCode == "nl-NL"){
+            languageCode = "en-US";
+        }
+        if(languageCode == "sv-SE"){
+            languageCode = "en-US";
+        }
+
+        var previewQueryString = "?lang=" + languageCode;
+
+        return encodeURI(baseurl + previewpage + previewQueryString);
+    };
+
+    gpii.firstDiscovery.showPanel = function (that) {
+        var panels = that.panels,
+            currentPanelNum = that.model.currentPanelNum,
+            selectorForCurrent = that.options.styles.currentPanel;
+
+        fluid.each(panels, function (panel, index) {
+            var toShow = currentPanelNum === (index + 1);
+            $(panel).toggleClass(selectorForCurrent, toShow);
+            if (toShow) {
+                var panelId = fluid.allocateSimpleId(panel);
+                that.events.onPanelShown.fire(panelId);
+            }
+        });
+    };
+    
+    /**
+     * setVoicingToggle
+     * This function disables the voice toggle button on the confirmation panel, congratulations and token panel. It enables the voice toggle on 
+     *     other panels in case the user goes back after seeing the confirmation screen. 
+     */
+    gpii.firstDiscovery.setVoicingToggle = function (that) {
+    	var CONFIRMATION_PANEL_ID=13;

```

Comment:
Use the DOM binder and configurable selectors rather than hard-coded numeric global ids